File: 141-linked-list-cycle/linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public boolean hasCycle(ListNode head) {

        if (head == null) return false;

        ListNode l = head;
        ListNode s = head;

        while (l.next != null && l.next.next != null) {
            l = l.next.next;
            s = s.next;

            if (s == l) return true;
        }

        return false;
        
        
    }
}

File: 141-linked-list-cycle/linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        if head == None:
            return False

        s = head
        l = head

        while l.next != None and l.next.next != None:
            l = l.next.next
            s = s.next

            if s == l:
                return True

        return False
        